src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -982,18 +974,30 @@ private void InitData()
         {
             if (_data is null)
             {
-                NodeType nodeType = NodeType;
-                object[] data = nodeType switch
-                {
-                    NodeType.Unknown => throw new InvalidOperationException($"Cannot resolve children of an {nameof(NodeType.Unknown)} node"),
-                    NodeType.Branch => new object[AllowBranchValues ? BranchesCount + 1 : BranchesCount],
-                    _ => new object[2]
-                };
-
-                // Only initialize the array if not already initialized.
-                Interlocked.CompareExchange(ref _data, data, null);
-                // Set NodeType after setting key as it alters code path to one that expects the key to be set.
-                NodeType = nodeType;
+                Initialize(NodeType);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void Initialize(NodeType nodeType)
```

Comment:
Can we make this a sub-method of InitData?

Reply:
https://github.com/NethermindEth/nethermind/pull/6661